appinventor/components/src/com/google/appinventor/components/runtime/util/FroyoUtil.java

```diff
@@ -125,4 +127,16 @@ public static WebViewClient getWebViewClient(final boolean ignoreErrors,
     return new FroyoWebViewClient(followLinks, ignoreErrors, form, component);
   }
 
+  /**
+   * Prior to SDK 9, java.io.IOException did not take a throwable as an argument.
+   * This function accepts a Throwable, calls toString and throws an IOException with
+   * just the string, which is supported.
+   *
+   * @param throwable the Throwable to re-throw (sort of)
+   */
+
+  public static void throwIOException(Throwable e) throws IOException {
+    throw new IOException(e.toString());
```

Comment:
So my comment in the chat was that we lose the stack trace for `e` at this point on Froyo and earlier. We should consider printing the stack trace so that we know where `e` was originally thrown.